### ISSUE TITLE
Use Ontoserver for terminology validation

### ISSUE THREAD

**Issue by znatty22** [kids-first/kf-model-fhir#116]
Rather than creating `CodeSystem` resources in our FHIR model for large ontologies, we should reference `CodeSystem`s in an external terminology server.  

The default terminology server that the IG publisher uses is: http://fhir3.healthintersections.com.au/ but this server does not have all of the ontologies we depend on. We should use Ontoserver (https://r4.ontoserver.csiro.au/fhir), developed by CSIRO Australia, which hosts HPO, NCIT, and other ontologies we need.

To do this, we must point the IG publisher to an Ontoserver instance. We could point to the public server or spin up our own Ontoserver, and then execute the following command:

```
java -jar org.hl7.fhir.publisher.jar -ig site_root/ig.ini -tx https://r4.ontoserver.csiro.au/fhir
```

To do:

- [x] Test the above locally to ensure it works
- [ ] ~~Update `run_publisher.sh` with `-tx https://r4.ontoserver.csiro.au/fhir`~~
- [x] Update .`circleci/config.yml` with `-tx https://r4.ontoserver.csiro.au/fhir`

**Comment by liberaliscomputing** [comment]
I talked to Alejandro and figured the following command tries to load all CodeSystems/ValueSets into output HTML files: 

```bash
[fhirmodel validate]|[java -jar org.hl7.fhir.publisher.jar] -ig site_root/ig.ini -tx https://r4.ontoserver.csiro.au/fhir
```
The HPO CodeSystem includes at least 20,000 concepts and is therefore huge. Instead, we need to pass in configuration parameters to the terminology server (See path-expansion-params [here](https://confluence.hl7.org/display/FHIR/Implementation+Guide+Parameters)). 

This means we need to add a parameter in our IG resource as follows:

```json
"parameter": [
  {
    "code": "copyrightyear",
    "value": "2020+"
  },
  {
    "code": "releaselabel",
    "value": "CI Build"
  },
  {
    "code": "path-expansion-params",
    "value": "input/pagecontent/Parameters-kfdrc-ontoserver.json"
  }
]
```

And `Parameters-kfdrc-ontoserver.json` should look something like this:

```json
{
  "resourceType": "Parameters",
  "id": "kfdrc-ontoserver",
  "parameter": [
    {
      "name": "count",
      "valueInteger": 1000
    }
  ]
}
```

**Comment by znatty22** [comment]
Thanks @liberaliscomputing so I tried the ^ but it looks like `path-expansion-params` is old. The FHIR resource docs for IG.parameters say there is a code called `expansion-parameters`. I tried 
```
{
        "code": "expansion-parameter",
        "value": "input/resources/terminology/Parameters-kfdrc-ontoserver.json"
}
```
And that worked. Also I don't think we need to refactor that code block for a couple reasons. One is that this `Parameters` FHIR resource isn't a FHIR resource that needs to be documented in the IG like the other ones. Second is, even if the `Parameters` did need to be documented, that code block doesn't actually remove any of the other parts of the IG configuration. It just grabs the resources from definitions and updates that section.

**Comment by znatty22** [comment]
@liberaliscomputing Can you make a PR for the changes listed in the issue description and also include the new IG configuration for the expansion parameters?

**Comment by liberaliscomputing** [comment]
@znatty22, Using the former returns one build error, saying:

```
ImplementationGuide.definition.parameter[25].value	error	Supressed messages file is not using the new format (see https://confluence.hl7.org/display/FHIR/Implementation+Guide+Parameters).
```

The official 4.0.1 IG uses `expansion-parameter` (https://www.hl7.org/fhir/codesystem-guide-parameter-code.html#guide-parameter-code-expansion-parameter) whereas the Confluence wiki (https://confluence.hl7.org/display/FHIR/Implementation+Guide+Parameters) uses `path-expansion-params`. Using `expansion-parameter` only raises one build error and passes validation while using the latter fails validation. We may want to report this somewhere?

**Comment by liberaliscomputing** [comment]
Under `"definition"` of IG, I added the following to `"parameter"`:

```json
{
  "code": "expansion-parameter",
  "value": "input/resources/terminology/Parameters-kfdrc-ontoserver.json"
}
```

and it still generates the following HTML file including all the HPO codes: [ValueSet-phenotype-codes.html.zip](https://github.com/kids-first/kf-model-fhir/files/4619001/ValueSet-phenotype-codes.html.zip)

To better understand the use of `"parameter"` in an [ImplementationGuide](https://www.hl7.org/fhir/implementationguide.html) resoure, I've looked into the [ValueSet](https://www.hl7.org/fhir/valueset-guide-parameter-code.html) of parameter code. It says when `"code"` is `"expansion-parameter"`, `"value"` should be a named parameter. Therefore, I created the following resource as `input/resources/terminology/Parameters-kfdrc-ontoserver.json`:

```json
{
  "resourceType": "Parameters",
  "id": "kfdrc-ontoserver",
  "parameter": [
    {
      "name": "count",
      "valueInteger": 1000
    }
  ]
}
```
and added the following to the IG file:

```json
{
  "code": "expansion-parameter",
  "value": "count"
}
```

However, it still generates the same HTML file including all HPO codes.
